File: app/src/main/java/com/seafile/seadroid2/framework/model/BaseModel.java
package com.seafile.seadroid2.framework.model;

import androidx.room.Ignore;

import com.seafile.seadroid2.config.Constants;
import com.seafile.seadroid2.enums.ItemPositionEnum;
import com.seafile.seadroid2.framework.db.entities.FileBackupStatusEntity;
import com.seafile.seadroid2.ui.data_migrate.DataMigrationActivity;

public class BaseModel {
    /**
     * <h3>data version control</h3>
     * <p>when v = 0, In some entities, this represents pre-migration legacy data.
     * eg. {@link FileBackupStatusEntity}.</p>
     * <br>
     * <p>
     * <b>notice</b>: Some values may have invalid or null, please check before using them.</p>
     * see also: {@link DataMigrationActivity}
     * <br>
     * <br>
     * <p>
     * when v = 1, it's new data after migration, and app version is >= 3.0.0.
     * </p>
     * <br>
     * <p><b>default is 1.<b/></p>
     */
    public int v = 1;


    /**
     * @see com.seafile.seadroid2.config.Constants.DataStatus
     */
    public int data_status = Constants.DataStatus.NORMAL;

    @Ignore
    public boolean is_checked = false;

    @Ignore
    public ItemPositionEnum item_position;
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/BaseModel.kt
package com.seafile.seadroid2.framework.model

import androidx.room.Ignore
import com.seafile.seadroid2.config.Constants
import com.seafile.seadroid2.enums.ItemPositionEnum
import com.seafile.seadroid2.framework.db.entities.FileBackupStatusEntity
import com.seafile.seadroid2.ui.data_migrate.DataMigrationActivity

/**
 * Controls data versioning for models that participate in migrations.
 *
 * When [v] == 0 legacy data may be incomplete, see [FileBackupStatusEntity] and [DataMigrationActivity].
 * When [v] == 1 the record has been migrated (app version >= 3.0.0). Default is 1.
 */
open class BaseModel {
    /**
     * Controls data versioning for models that participate in migrations.
     *
     * When [v] == 0 legacy data may be incomplete, see [FileBackupStatusEntity] and [DataMigrationActivity].
     * When [v] == 1 the record has been migrated (app version >= 3.0.0). Default is 1.
     */
    @JvmField
    var v: Int = 1

    /**
     * @see com.seafile.seadroid2.config.Constants.DataStatus
     */
    @JvmField
    var data_status: Int = Constants.DataStatus.NORMAL

    @Ignore
    @JvmField
    var is_checked: Boolean = false

    @Ignore
    @JvmField
    var checkable: Boolean = true

    @Ignore
    @JvmField
    var item_position: ItemPositionEnum? = null
}
